python/cudf/cudf/core/frame.py

```diff
@@ -1322,8 +1322,7 @@ def _merge(
             for i, col in enumerate(by):
                 to_sort[i] = col
             inds = to_sort.argsort()
-            for col in to_return.columns:
-                to_return._data[col] = to_return._data[col].take(inds)
+            to_return._data = to_return.take(inds)._data
```

Comment:
This looks like we still gather the RangeIndex and then throw it away.

Reply:
Yes. I'm not sure how to avoid this without modifying `take`, or calling `take` on the columns individually like it was.

Reply:
Although it might not be hard to extend `take` in this case, it probably gets a `view` and could get a `data_view`.

Reply:
I'm thinking we add a kwarg to `take` of `ignore_index=False`

Reply:
Already done :)